ctests/CMakeLists.txt

```
@@ -0,0 +1,106 @@
+if (NOT DEFINED HPC_ACCOUNT)
+  message(FATAL_ERROR "HPC_ACCOUNT must be set")
+  return()
+endif()
+
+# Set ICSDIR_ROOT
+set_from_env_or_default(ICSDIR_ROOT ICSDIR_ROOT "")
+if (NOT DEFINED ICSDIR_ROOT)
+  message(FATAL_ERROR "ICSDIR_ROOT must be set")
+  return()
+endif()
+
+# Set STAGED_TESTS_DIR
+set_from_env_or_default(STAGED_TESTS_DIR STAGED_TESTS_DIR "")
+if (NOT DEFINED STAGED_TESTS_DIR)
+  message(FATAL_ERROR "STAGED_TESTS_DIR must be set")
+  return()
+endif()
+
+message(STATUS "gw: global-workflow baselines will be used from: '${HOMEgfs}'")
+message(STATUS "gw: global-workflow tests will be run at: '${RUNTESTS}'")
+message(STATUS "gw: global-workflow tests will use the allocation: '${HPC_ACCOUNT}'")
+message(STATUS "gw: global-workflow tests will use ICSDIR_ROOT: '${ICSDIR_ROOT}'")
+message(STATUS "gw: global-workflow tests will use staged data from:  '${STAGED_TESTS_DIR}'")
+
+# Prepare test scripts
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/scripts/setup.sh.in
+               ${CMAKE_CURRENT_BINARY_DIR}/scripts/setup.sh @ONLY)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/scripts/stage.sh.in
+               ${CMAKE_CURRENT_BINARY_DIR}/scripts/stage.sh @ONLY)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/scripts/execute.sh.in
+               ${CMAKE_CURRENT_BINARY_DIR}/scripts/execute.sh @ONLY)
+# TODO - Validation step add its CMake Teplate for its bash wrapper script
+# configure_file(${CMAKE_CURRENT_SOURCE_DIR}/scripts/validate.sh.in
+#                ${CMAKE_CURRENT_BINARY_DIR}/scripts/validate.sh @ONLY)
+
+function(AddJJOBTest)
+
+  set(prefix     ARG)
+  set(novals     NOTRAPFPE NOVALGRIND)
+  set(singlevals CASE JOB TEST_DATE)
+  set(multivals  TEST_DEPENDS)
+
+  cmake_parse_arguments(${prefix}
+                       "${novals}" "${singlevals}" "${multivals}"
+                        ${ARGN})
+
+  set(TEST_NAME ${ARG_CASE}_${ARG_JOB})
+  set(CASE_PATH ${HOMEgfs}/ci/cases/pr)
+  set(CASE_YAML ${CASE_PATH}/${ARG_CASE}.yaml)
+
+  add_test(NAME test_${TEST_NAME}_setup
+    COMMAND ./setup.sh ${TEST_NAME} ${CASE_YAML} ${ARG_TEST_DATE}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/scripts)
+  set_tests_properties(test_${TEST_NAME}_setup PROPERTIES LABELS "${ARG_CASE};${ARG_JOB}")
+    
+  add_test(NAME test_${TEST_NAME}_stage
+      COMMAND ./stage.sh ${TEST_NAME} ${ARG_TEST_DATE}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/scripts)
+  set_tests_properties(test_${TEST_NAME}_stage PROPERTIES DEPENDS test_${TEST_NAME}_setup LABELS "${ARG_CASE};${ARG_JOB}")
+  
+  add_test(NAME test_${TEST_NAME}_execute
+      COMMAND ./execute.sh ${TEST_NAME} ${ARG_JOB} ${ARG_TEST_DATE}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/scripts)
+  set_tests_properties(test_${TEST_NAME}_execute PROPERTIES DEPENDS test_${TEST_NAME}_stage LABELS "${ARG_CASE};${ARG_JOB}")
+ 
+  # TODO - This is a stub for the validation step
+  add_test(NAME test_${TEST_NAME}_validate
+      COMMAND bash -c "${HOMEgfs}/ctests/scripts/validate.sh")
```

Why not just do the same for `validate.sh` from the templated `validate.sh.in`?
If not, atleast this"
```suggestion
      COMMAND ./validate.sh
      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/scripts))
```